Description: Retry maas-enlist without power parameters on failure
 Older versions of maas-enlist (such as in Precise) do not support the
 new power parameters switches and fail when they are provided, so on
 failure retry without them.
Forwarded: https://code.launchpad.net/~racb/maas/fix-ipmi-enlistment/+merge/128914
Author: Robie Basak <robie.basak@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/1064922
Last-Update: 20121010

--- a/contrib/preseeds_v2/enlist_userdata
+++ b/contrib/preseeds_v2/enlist_userdata
@@ -182,7 +182,10 @@
    fi
    power_params=$(maas-ipmi-autodetect --configdir "$IPMI_CONFIG_D" ${pargs} --commission-creds) &&
      [ -n "${power_params}" ] && power_params=${power_params%.} && power_type=ipmi
-   maas-enlist --serverurl "$url" ${host:+--hostname "${host}"} ${power_params:+--power-params "${power_params}" --power-type "${power_type}"}>/tmp/enlist.out
+   # Try maas-enlist without power parameters on failure for older versions of
+   # maas-enlist without power parameter support
+   maas-enlist --serverurl "$url" ${host:+--hostname "${host}"} ${power_params:+--power-params "${power_params}" --power-type "${power_type}"}>/tmp/enlist.out ||\
+   maas-enlist --serverurl "$url" ${host:+--hostname "${host}"} >/tmp/enlist.out
    if [ $? -eq 0 ]; then
       msg="successfully enlisted to '$url'"
       [ -n "$host" ] && msg="$msg with hostname '$host'" ||
